pkg/tun/device_windows.go

```diff
@@ -102,7 +112,40 @@ func (t *Device) setDNS(ctx context.Context, server net.IP, domains []string) (e
 	if err = luid.SetDNS(family, []net.IP{server}, domains); err != nil {
 		return err
 	}
-	_ = dexec.CommandContext(ctx, "ipconfig", "/flushdns").Run()
+
+	// On some systems, SetDNS isn't enough to allow the domains to be resolved, and the network adapter's domain has to be set explicitly.
+	// It's actually way easier to do this via powershell than any system calls that can be run from go code
+	domain := ""
+	if len(domains) > 0 {
+		// Quote the domain to prevent powershell injection
+		domain = shellquote.ShellArgsString([]string{strings.TrimSuffix(domains[0], ".")})
+	}
+	// It's apparently well known that WMI queries can hang under various conditions, so we add a timeout here to prevent hanging the daemon
+	// Fun fact: terminating the context that powershell is running in will not stop a hanging WMI call (!) perhaps because it is considered uninterruptible
+	pshScript := fmt.Sprintf(`
+$job = Get-WmiObject Win32_NetworkAdapterConfiguration -filter "interfaceindex='%d'" -AsJob | Wait-Job -Timeout 30
+if ($job.State -ne 'Completed') {
+	throw "timed out getting network adapter after 30 seconds."
+}
+$obj = $job | Receive-Job
+$job = Invoke-WmiMethod -InputObject $obj -Name SetDNSDomain -ArgumentList "%s" -AsJob | Wait-Job -Timeout 30
+if ($job.State -ne 'Completed') {
+	throw "timed out setting network adapter DNS Domain after 30 seconds."
+}
+$job | Receive-Job
+`, t.interfaceIndex, domain)
+	cmd := dexec.CommandContext(ctx, "powershell.exe", "-NoProfile", "-NonInteractive", pshScript)
+	cmd.DisableLogging = true // disable chatty logging
+	dlog.Debugf(ctx, "Calling powershell's SetDNSDomain %q", domain)
+	if err := cmd.Run(); err != nil {
+		// Log the error, but don't actually fail on it: This is all just a fallback for SetDNS, so the domains might actually be working
+		dlog.Errorf(ctx, "Failed to set NetworkAdapterConfiguration DNS Domain: %v. Will proceed, but namespace mapping might not be functional.", err)
+	}
```

Comment:
gross gross gross :)

I'm not looking at this too closely, but it would be good to include references in those comments ("apparently well-known"), and maybe be more specific that "some systems".

Reply:
> would be good to include references in those comments ("apparently well-known"),

Good point, I'll add some links to the comment

> and maybe be more specific that "some systems".

This I can't do :( unfortunately "some systems" means "CircleCI but not Jose's windows computer"